x-pack/plugins/lens/public/indexpattern_datasource/dimension_panel/time_scaling.tsx

```diff
@@ -0,0 +1,187 @@
+              color="text"
+              onClick={() => {
+                setState(
+                  mergeLayer({
+                    state,
+                    layerId,
+                    newLayer: {
+                      ...state.layers[layerId],
+                      columns: {
+                        ...state.layers[layerId].columns,
+                        [columnId]: {
+                          ...selectedColumn,
+                          timeScale: DEFAULT_TIME_SCALE,
+                        },
+                      },
+                    },
+                  })
+                );
+              }}
+            >
+              {i18n.translate('xpack.lens.indexPattern.timeScale.enableTimeScale', {
+                defaultMessage: 'Show as rate',
+              })}
+            </EuiLink>
+          </EuiText>
+        </EuiPopover>
+      </EuiText>
+    );
+  }
+
+  return (
+    <EuiFormRow
+      display="columnCompressed"
+      fullWidth
+      label={i18n.translate('xpack.lens.indexPattern.timeScale.label', {
+        defaultMessage: 'Normalize by unit',
+      })}
+    >
+      <EuiFlexGroup gutterSize="s" alignItems="center">
+        <EuiFlexItem>
+          <EuiSelect
+            compressed
+            options={Object.entries(unitSuffixesLong).map(([unit, text]) => ({
+              value: unit,
+              text,
+            }))}
+            data-test-subj="indexPattern-time-scaling-unit"
+            value={selectedColumn.timeScale}
+            onChange={(e) => {
+              setState(
+                mergeLayer({
+                  state,
+                  layerId,
+                  newLayer: {
+                    ...state.layers[layerId],
+                    columns: {
+                      ...state.layers[layerId].columns,
+                      [columnId]: {
+                        ...selectedColumn,
+                        timeScale: e.target.value as TimeScaleUnit,
+                      },
+                    },
+                  },
+                })
+              );
+            }}
+          />
+        </EuiFlexItem>
+        {selectedOperation.timeScalingMode === 'optional' && (
+          <EuiFlexItem grow={false}>
+            <EuiButtonIcon
+              data-test-subj="indexPattern-time-scaling-remove"
+              color="danger"
+              aria-label={i18n.translate('xpack.lens.timeScale.removeLabel', {
+                defaultMessage: 'Remove normalizing by unit',
+              })}
+              onClick={() => {
+                setState(
+                  mergeLayer({
+                    state,
+                    layerId,
+                    newLayer: {
+                      ...state.layers[layerId],
+                      columns: {
+                        ...state.layers[layerId].columns,
+                        [columnId]: {
+                          ...selectedColumn,
+                          timeScale: undefined,
```

Comment:
If timeScale were part of params, you'd be allowed to use the `updateColumnParam` helper from `layer_helpers`. Would be simpler than writing all this!

Unfortunately this idea conflicts with my idea of using a new `updateLayer` function instead of `setState`- we can only pick one of these for simplification.

Reply:
I think `updateLayer` makes sense here because we anyway have to adjust the label as well based on your other comment